File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInventoryContents.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

import java.util.Objects;

public class PSPInventoryContents implements PlayerStateProperty<ItemStack[]> {
    @Override
    public String getConfigName() {
        return "inventory_contents";
    }

    @Override
    public ItemStack[] getDefaultSerializedProperty() {
        return new ItemStack[0];
    }

    @Override
    public ItemStack[] getSerializedPropertyFromPlayer(Player player) {
        return player.getInventory().getContents();
    }

    @Override
    public void restoreProperty(Player player, ItemStack[] contents) {
        player.getInventory().setContents(contents);
    }

    @Override
    @SuppressWarnings("SuspiciousToArrayCall")
    public ItemStack[] getValueFromConfig(ConfigurationSection config) {
        return Objects.requireNonNull(config.getList(getConfigName())).toArray(new ItemStack[0]);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInventoryContents.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.noxet.noxetserver.playerstate.PlayerStateProperty

import scala.jdk.CollectionConverters.*

object PSPInventoryContents extends PlayerStateProperty[Array[ItemStack]]:
  override def getConfigName: String = "inventory_contents"

  override def getDefaultSerializedProperty: Array[ItemStack] = Array.empty

  override def getSerializedPropertyFromPlayer(player: Player): Array[ItemStack] =
    player.getInventory.getContents

  override def restoreProperty(player: Player, contents: Array[ItemStack]): Unit =
    player.getInventory.setContents(contents)

  override def getValueFromConfig(config: ConfigurationSection): Array[ItemStack] =
    config.getList(getConfigName).asScala.toArray